File: jee/java/src/main/java/fables/kotlin/jee/java/rest/KittenRest.java
package fables.kotlin.jee.java.rest;

import com.fasterxml.jackson.annotation.JsonProperty;
import fables.kotlin.jee.java.Kitten;

/**
 * Kitten data transfer object.
 *
 * @author Zeljko Trogrlic
 */
public class KittenRest
    implements Kitten {

  private final String name;
  private final int    cuteness;

  public String getName() {
    return this.name;
  }

  public int getCuteness() {
    return this.cuteness;
  }

  /**
   * Default constructor.
   * On deserialization, properties cannot be mapped without @JsonProperty annotations.
   */
  public KittenRest(
      @JsonProperty("name") String name,
      @JsonProperty("cuteness") int cuteness
  ) {
    this.name = name;
    this.cuteness = cuteness;
  }
}

File: jee/converted/src/main/kotlin/fables/kotlin/jee/rest/KittenRest.kt
package fables.kotlin.jee.rest

import com.fasterxml.jackson.annotation.JsonProperty
import fables.kotlin.jee.Kitten

/**
 * Kitten data transfer object.
 * On deserialization, default constructor properties cannot be mapped without @JsonProperty annotations.
 * @author Zeljko Trogrlic
 */
data class KittenRest(
        @param:JsonProperty("name") override val name: String,
        @param:JsonProperty("cuteness") override val cuteness: Int
) : Kitten
